core/base/src/TROOT.cxx

```diff
@@ -2136,20 +2124,10 @@ void TROOT::ReadGitInfo()
 
    TString gitinfo = "gitinfo.txt";
    char *filename = 0;
-#ifdef ROOTETCDIR
-   filename = gSystem->ConcatFileName(ROOTETCDIR, gitinfo);
-#else
-   TString etc = gRootDir;
-#ifdef WIN32
-   etc += "\\etc";
-#else
-   etc += "/etc";
-#endif
 #if defined(R__MACOSX) && (TARGET_OS_IPHONE || TARGET_IPHONE_SIMULATOR)
-   // on iOS etc does not exist and gitinfo resides in $ROOTSYS
-   etc = gRootDir;
-#endif
-   filename = gSystem->ConcatFileName(etc, gitinfo);
+   filename = gSystem->ConcatFileName(TROOT::GetRootSys(), gitinfo);
```

Comment:
This didn't make sense to me in the first place. I simply tried to not change what de code did so that I didn't introduce any new behaviour.

The main reason for thinking that it is strange is that if "etc does not exist on ios", then I would have expected this #ifdef to appear every place where ROOTETCDIR is used. This is not the case. It appears - if I remember correctly - twice in the whole source tree. For all the other places ROOTETCDIR is used also for ios.

If ios does not have etc I would have expected that for ios configure/cmake would default to --etcdir='${prefix}' and then ROOTETCDIR would correctly point to rootsys already and no #ifdefs would be needed.

I agree the code as it is written is strange - but I didn't want to just assume the code was wrong and "fix" it in case I had the wrong idea of what was the intended behaviour.